### PR TITLE
Support String Repeat func C backend 

### LINKED ISSUE #1101: Implement ``visit_StringConcat`` method in C backend
On executing `lpython --show-c integration_tests/test_str_01.py` I get,

```bash
LCompilersException: visit_StringConcat() not implemented
```

So we need to implement `visit_StringConcat` for C backend in the following class,

https://github.com/lcompilers/lpython/blob/8e47b95d9b4b6b0e40877cdd0d866ae9b172f3fb/src/libasr/codegen/asr_to_c.cpp#L34-L46

### PR DESCRIPTION
**Pull request by ronnuriel**
closes  #1101

### CODE REVIEW

File: src/libasr/codegen/asr_to_c_cpp.h [hunk 1]
```diff
@@ -1137,6 +1138,30 @@ R"(#include <stdio.h>
...
+    void visit_StringConcat(const ASR::StringConcat_t &x) {
+    if (is_strcat_util_func_added){
+        additional_functions =
+            "char *concat (const char *str1, const char *str2) {\n"
+            "\n"
+            "size_t len1 = strlen (str1);         /* get lenghts */\n"
+            "size_t len2 = strlen (str2);\n"
+            "\n"
+            "char * s = malloc (len1 + len2 + 1); /* allocate s and concat with ' ' */\n"
+            "memcpy (s, str1, len1);\n"
```
Comment (Smit-create) [inline src/libasr/codegen/asr_to_c_cpp.h:1150]: I was thinking if we can add all the helper functions like this or like in #1109 in some other file and include that header file.  Something similar to `lfortran_intrinsics.h` or do we want some other new file for this? @certik @czgdp1807

Comment (czgdp1807) [reply src/libasr/codegen/asr_to_c_cpp.h:1150]: I agree all these functions fit well in `lfortran_intrinsics.h`. That way no need to define dynamically in C backend.

File: src/libasr/codegen/asr_to_c_cpp.h [hunk 2]
```diff
@@ -1137,6 +1137,45 @@ R"(#include <stdio.h>
...
+    std::string left_expr = src;
+     self().visit_expr(*x.m_right);
+    std::string right_expr = src;
+     src = "concat (" + left_expr + "," + right_expr +")";
+}
+
+    void visit_StringSection(const ASR::StringSection_t &x) {
+        this->visit_expr(*x.m_arg);
+        std::string str = src;
+        std::string start, end, step;
```
Comment (Smit-create) [inline src/libasr/codegen/asr_to_c_cpp.h:1151]: I have handled this in #1109, we have a function `_lfortran_str_slice` already defined in `lfortran_intrinsic`

File: src/libasr/codegen/asr_to_c_cpp.h [hunk 3]
```diff
@@ -1137,6 +1137,45 @@ R"(#include <stdio.h>
         last_expr_precedence = 16;
     }
 
+    void visit_StringConcat(const ASR::StringConcat_t &x) {
+    self().visit_expr(*x.m_left);
+    std::string left_expr = src;
+     self().visit_expr(*x.m_right);
+    std::string right_expr = src;
```
Comment (Smit-create) [inline src/libasr/codegen/asr_to_c_cpp.h:1144]: For `StringConcat`, `lfortran_intrinsic.c` already has a function `_lfortran_strcat`. You can directly use that.

**Comment by Smit-create** [comment]
@ronnuriel  Let me know if you are busy, I can complete this.

**Comment by ronnuriel** [comment]
@Smit-create yes please, Im working on the goto issue now..

**Comment by Smit-create** [comment]
This is ready for review @certik @czgdp1807

**Comment by Smit-create** [comment]
@certik @czgdp1807 Should we merge this?

**Comment by Smit-create** [comment]
> @Smit-create If you can update this with main and complete it then sure we can merge this.

Done!

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**